x/crosschain/keeper/evm_hooks.go

```diff
@@ -72,12 +74,25 @@ func (k Keeper) ProcessLogs(ctx sdk.Context, logs []*ethtypes.Log, emittingContr
 				return err
 			}
 		}
+		// We were able to parse the ZRC20 withdrawal event. However, we were unable to process it as the information was incorrect
+		// This means that there are some funds locked in the contract which cannot have an outbound , this can be a candidate for a refund
+		// TODO : Consider returning error or auto refunding the funds to the user
+		if eventWithdrawal != nil {
```

Comment:
We want to error out when the Withdrawal event needs to be processed, but the event contains invalid input from user such as unsupported address. 

Currently k.ParseZRC20WithdrawalEvent() returns non-nil error when the Withdrawal event needs to be processed **and** is correct. 

We need to have a condition here that says the Withdrawal event needs to be processed and it needs to be correct, otherwise error out so the EVM tx will fail. 

Thinking either `k.ParseZRC20WithdrawalEvent` returns an additional field that indicates this is a relevant Withdrawal event (the criteria is--the emitting contract is one of the ZRC20s in the foreign coin list).